File: app/src/main/java/com/asvsdfer/bjirmndf/ui/MainActivity.java
package com.asvsdfer.bjirmndf.ui;

import android.util.Log;
import android.view.KeyEvent;
import android.view.WindowManager;
import android.widget.TextView;

import androidx.fragment.app.Fragment;
import androidx.viewpager2.widget.ViewPager2;

import com.asvsdfer.bjirmndf.R;
import com.asvsdfer.bjirmndf.api.RetrofitManager;
import com.asvsdfer.bjirmndf.base.BaseActivity;
import com.asvsdfer.bjirmndf.base.ObserverManager;
import com.asvsdfer.bjirmndf.model.BaseModel;
import com.asvsdfer.bjirmndf.model.ConfigEntity;
import com.asvsdfer.bjirmndf.ui.adapter.BaseFragmentAdapter;
import com.asvsdfer.bjirmndf.ui.fragment.MainFragment;
import com.asvsdfer.bjirmndf.util.SharePreferencesUtil;
import com.asvsdfer.bjirmndf.util.StatusBarUtil;
import com.asvsdfer.bjirmndf.util.ToastUtil;
import com.flyco.tablayout.CommonTabLayout;
import com.flyco.tablayout.listener.CustomTabEntity;
import com.flyco.tablayout.listener.OnTabSelectListener;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.schedulers.Schedulers;

public class MainActivity extends BaseActivity {

    private ViewPager2 homeViewPager;
    private CommonTabLayout tabLayout;

    private long exitTime = 0;
    private List<Fragment> mFragments = new ArrayList<>();
    private String[] mTitles = {"首页", "精选", "我的"};
    private int[] uncheckedIcon = {R.drawable.shouye2, R.drawable.jingxuan2, R.drawable.wode2};
    private int[] checkedIcon = {R.drawable.shouye1, R.drawable.jingxuan1, R.drawable.wode1};
    private ArrayList<CustomTabEntity> customTabEntities;

    @Override
    public int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    public void initListener() {

    }

    @Override
    public void initData() {
        StatusBarUtil.setTransparent(this, false);
        if (SharePreferencesUtil.getBool("NO_RECORD")) {
            getWindow().addFlags(WindowManager.LayoutParams.FLAG_SECURE);
        }
//        getP().login();
        homeViewPager = findViewById(R.id.home_view_pager);
        tabLayout = findViewById(R.id.tab_layout);
        customTabEntities = new ArrayList<>();
        homeViewPager.setUserInputEnabled(false);
        for (int i = 0; i < 3; i++) {
            int position = i;
            CustomTabEntity customTabEntity = new CustomTabEntity() {
                @Override
                public String getTabTitle() {
                    return mTitles[position];
                }

                @Override
                public int getTabSelectedIcon() {
                    return checkedIcon[position];
                }

                @Override
                public int getTabUnselectedIcon() {
                    return uncheckedIcon[position];
                }
            };
            customTabEntities.add(customTabEntity);
        }
        tabLayout.setTabData(customTabEntities);
        tabLayout.setOnTabSelectListener(new OnTabSelectListener() {
            @Override
            public void onTabSelect(int position) {
                homeViewPager.setCurrentItem(position, false);
            }

            @Override
            public void onTabReselect(int position) {

            }
        });
        mFragments.add(new MainFragment());
        mFragments.add(new MainFragment());
        mFragments.add(new MainFragment());

        homeViewPager.setAdapter(new BaseFragmentAdapter(getSupportFragmentManager(), getLifecycle(), mFragments));
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_BACK
                && event.getAction() == KeyEvent.ACTION_DOWN) {
            if ((System.currentTimeMillis() - exitTime) > 2000) {
                ToastUtil.showShort("再按一次退出程序");
                exitTime = System.currentTimeMillis();
            } else {
                finish();
                System.exit(0);
            }
            return true;
        }
        return super.onKeyDown(keyCode, event);
    }
}

File: app/src/main/java/com/qwsdbsdvdsnd/dfmcvndfhdfsdfwe/ui/activity/MainActivity.kt
package com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.ui.activity

import android.os.Bundle
import android.view.KeyEvent
import android.view.WindowManager
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.R
import com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.api.NetSimple
import com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.mode.BaseModel
import com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.mode.ConfigModel
import com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.mode.LoginModel
import com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.ui.adapter.FragmentAdapter
import com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.ui.fragment.HomeFragment
import com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.ui.fragment.ListFragment
import com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.ui.fragment.WodeFragment
import com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.util.SPUtil
import com.qwsdbsdvdsnd.dfmcvndfhdfsdfwe.util.StatusBarUtil
import com.flyco.tablayout.listener.CustomTabEntity
import com.flyco.tablayout.listener.OnTabSelectListener
import kotlinx.android.synthetic.main.activity_login_page.*
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.*
import kotlin.collections.ArrayList
import kotlin.system.exitProcess

class MainActivity : AppCompatActivity() {

    private var phone = ""
    private var ip = ""
    private var exitTime: Long = 0
    private var mFragments: MutableList<Fragment> = ArrayList()
    private var mTitles = arrayOf("首页", "精选", "我的")
    private val uncheckedIcon =
        intArrayOf(R.drawable.vnxet, R.drawable.kdtysfgh, R.drawable.lpuikd)
    private val checkedIcon = intArrayOf(R.drawable.lpiukf, R.drawable.hjbnd, R.drawable.wrgxfm)
    private var customTabEntities: ArrayList<CustomTabEntity>? = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        StatusBarUtil.setTransparent(this, false)
        setContentView(R.layout.activity_main)
        logins()
        customTabEntities = ArrayList()
        home_view_pager.isUserInputEnabled = false
        for (i in 0..2) {
            val customTabEntity: CustomTabEntity = object : CustomTabEntity {
                override fun getTabTitle(): String {
                    return mTitles[i]
                }

                override fun getTabSelectedIcon(): Int {
                    return checkedIcon[i]
                }

                override fun getTabUnselectedIcon(): Int {
                    return uncheckedIcon[i]
                }
            }
            customTabEntities!!.add(customTabEntity)
        }
        tab_layout.setTabData(customTabEntities)
        tab_layout.setOnTabSelectListener(object : OnTabSelectListener {
            override fun onTabSelect(position: Int) {
                home_view_pager.setCurrentItem(position, false)
            }

            override fun onTabReselect(position: Int) {}
        })
        mFragments.add(HomeFragment())
        mFragments.add(ListFragment())
        mFragments.add(WodeFragment())

        home_view_pager.adapter = FragmentAdapter(supportFragmentManager, lifecycle, mFragments)
    }

    private fun logins(){
        phone = SPUtil.getString("phone")
        ip = SPUtil.getString("ip")
        val apiService = NetSimple.getRetrofitManager().apiService
        apiService.logins(phone, ip).enqueue(object : Callback<BaseModel<LoginModel>> {
            override fun onFailure(call: Call<BaseModel<LoginModel>>, t: Throwable) {
                t.printStackTrace()
            }

            override fun onResponse(
                call: Call<BaseModel<LoginModel>>,
                response: Response<BaseModel<LoginModel>>
            ) {

            }
        })
    }

    override fun onResume() {
        super.onResume()
        getConfig()
    }

    private fun getConfig() {
        val apiService = NetSimple.getRetrofitManager().apiService  //传入之前定义的接口

        //Get请求并处理结果
        apiService.getValue("VIDEOTAPE").enqueue(object : Callback<BaseModel<ConfigModel>> {
            override fun onFailure(call: Call<BaseModel<ConfigModel>>, t: Throwable) {
                t.printStackTrace()
            }

            override fun onResponse(
                call: Call<BaseModel<ConfigModel>>,
                response: Response<BaseModel<ConfigModel>>
            ) {
                val model = response.body() ?: return
                val configModel = model.data
                if (configModel != null) {
                    SPUtil.saveBool(
                        "NO_RECORD",
                        !configModel.videoTape.equals("0")
                    )
                    if (SPUtil.getBool("NO_RECORD")) {
                        window.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
                    }
                }
            }
        })
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent?): Boolean {
        if (keyCode == KeyEvent.KEYCODE_BACK
            && event!!.action == KeyEvent.ACTION_DOWN
        ) {
            if (System.currentTimeMillis() - exitTime > 2000) {
                Toast.makeText(this@MainActivity, "再按一次退出程序", Toast.LENGTH_SHORT).show()
                exitTime = System.currentTimeMillis()
            } else {
                finish()
                exitProcess(0)
            }
            return true
        }
        return super.onKeyDown(keyCode, event)
    }
}